src/vg_active_segment.erl

```diff
@@ -315,8 +327,11 @@ maybe_roll(_, State) ->
     State.
 
 %% Append encoded log to segment
-update_log(Record, #state{log_fd=LogFile}) ->
-    ok = file:write(LogFile, Record).
+update_log(Id, [<<_:64/signed-integer, Rest/binary>>, Record], #state{log_fd=LogFile}) ->
```

Comment:
`Rest` feels like a weird variable name here, maybe `Batch`?

Reply:
Eh, it actually isn't the batch here, that is `Record`. I don't like this code but not sure how to get rid of it. The issue is that the encode returns a iolist and not a flat binary, so we can't just match on the Id at the beginning to remove it.

I may instead need to remove Id (which is set to 0 when not set) from the `encode_record_batch` return so that the Id doesn't have to be dropped in `update_log` at all.